src/node_crypto.cc

```diff
@@ -5572,6 +5591,65 @@ void RandomBytes(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void RandomBytesBuffer(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!Buffer::HasInstance(args[0])) {
+    return env->ThrowTypeError("buf must be a Buffer");
+  }
+
+  const int64_t len = Buffer::Length(args[0]);
+
+  if (!args[1]->IsUint32()) {
+    return env->ThrowTypeError("offset must be a number >= 0");
+  }
+
+  if (!args[2]->IsUint32()) {
+    return env->ThrowTypeError("size must be a number >= 0");
+  }
+
+  int64_t offset = args[1]->IntegerValue();
+  int64_t size = args[2]->IntegerValue();
+
+  if (offset < 0 || offset > Buffer::kMaxLength)
+    return env->ThrowRangeError("offset is not a valid Smi");
```

Comment:
"Smi" is a bit too jargon-y, not many people are going to know what it means.  (As well, -1 is a valid SMI.)